js/src/forum/states/GlobalSearchState.js

```diff
@@ -7,7 +7,7 @@ export default class GlobalSearchState extends SearchState {
   }
 
   getValue() {
-    if (this.value === undefined || !this.value.length) {
+    if (this.value == undefined || !this.value.length) {
```

Comment:
Isn't this basically `!this.value` now? If `this.value` is `0` it'll `== undefined`

Reply:
We could just return `this.value || this.getInitialSearch() || ''` for simplicity?

Reply:
Yeah I think that's the same. `!!'' === false` so.

Reply:
Although actually... What happens if an initial search is present, but then the search is cleared? With `this.value || this.getInitialSearch() || ''`, `getValue()` would still return that initial value. I think that's why it checked for undefined but not null.

Reply:
The previous null check I added so it doesn't TypeError in case it actually is null when we check for the length. 
I don't think it can become null. Let's check for the empty string explicitly instead though.